src/content/en/2021/javascript.md

```diff
@@ -1,20 +1,1543 @@
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=709009883&format=interactive",
+  sheets_gid="183546956",
+  sql_file="ajax_request_per_page.sql"
+  )
+}}
+
+
+If we look at the long tail and check the number of async requests for the 100th percentile, the difference in the devices is quite remarkable.
+
+<figure markdown>
+<table>
+  <tr>
+   <td><strong><em>percentile</em></strong>
+   </td>
+   <td>desktop
+   </td>
+   <td>mobile
+   </td>
+  </tr>
+  <tr>
+   <td><p style="text-align: right">
+10</p>
+
+   </td>
+   <td><p style="text-align: right">
+2</p>
+
+   </td>
+   <td><p style="text-align: right">
+2</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td><p style="text-align: right">
+25</p>
+
+   </td>
+   <td><p style="text-align: right">
+3</p>
+
+   </td>
+   <td><p style="text-align: right">
+3</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td><p style="text-align: right">
+50</p>
+
+   </td>
+   <td><p style="text-align: right">
+4</p>
+
+   </td>
+   <td><p style="text-align: right">
+4</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td><p style="text-align: right">
+75</p>
+
+   </td>
+   <td><p style="text-align: right">
+8</p>
+
+   </td>
+   <td><p style="text-align: right">
+8</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td><p style="text-align: right">
+90</p>
+
+   </td>
+   <td><p style="text-align: right">
+15</p>
+
+   </td>
+   <td><p style="text-align: right">
+16</p>
+
+   </td>
+  </tr>
+  <tr>
+   <td><p style="text-align: right">
+100</p>
+
+   </td>
+   <td><p style="text-align: right">
+623</p>
+
+   </td>
+   <td><p style="text-align: right">
+867</p>
+
+   </td>
+  </tr>
+</table>
+
+<figcaption>{{ figure_link(caption="Distribution of the number of asynchronous requests made per page.", sheets_gid="183546956", sql_file="ajax_request_per_page.sql") }}</figcaption>
+</figure>
+
+The most asynchronous requests made by a desktop page is 623, which is eclipsed by the biggest mobile page, which makes 867 asynchronous requests!
+
+An alternative to the asynchronous AJAX requests are the synchronous. Rather than passing a request to a callback, they block the main thread until the request completes.
+
+However, this practice is [discouraged](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests#:~:text=Note%3A%20Starting%20with%20Gecko%2030.0%20(Firefox%2030.0%20/%20Thunderbird%2030.0%20/%20SeaMonkey%202.27)%2C%20Blink%2039.0%2C%20and%20Edge%2013%2C%20synchronous%20requests%20on%20the%20main%20thread%20have%20been%20deprecated%20due%20to%20their%20negative%20impact%20on%20the%20user%20experience) due to the potential for poor performance and user experiences, and many browsers already warn about the usage.  It would be intriguing to see how many pages still use synchronous AJAX requests.
```

Comment:
```suggestion
However, this practice is [discouraged](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests#:~:text=Note%3A%20Starting%20with%20Gecko%2030.0%20(Firefox%2030.0%20/%20Thunderbird%2030.0%20/%20SeaMonkey%202.27)%2C%20Blink%2039.0%2C%20and%20Edge%2013%2C%20synchronous%20requests%20on%20the%20main%20thread%20have%20been%20deprecated%20due%20to%20their%20negative%20impact%20on%20the%20user%20experience) due to the potential for poor performance and user experiences, and many browsers already warn about the usage. It would be intriguing to see how many pages still use synchronous AJAX requests.
```

Reply:
There is a little inconsistency in the spacing here. You can decide if you want to use one or two spaces for new sentences.